nautobot/dcim/tables/devices.py

```diff
@@ -783,7 +780,7 @@ class Meta(DeviceComponentTable.Meta):
             "device",
             "name",
             "label",
-            "status",
+            "installed_device__status",
```

Comment:
Is this something that we could build a pylint rule for?

Reply:
I'm not sure offhand how we would spell out such a rule, TBH.

Reply:
Fair enough, if I could bother you a bit more, can you explain what this does? Like the why and what of the PR.

Reply:
Instead of manually defining an HTML template for rendering this column (which, like any TemplateColumn, receives `record` as a context variable, and in this case manually evaluates `record.installed_device.status` as part of the template), we now:

1. use a ColoredLabelColumn, which is a Nautobot-specific class that knows how to render Status, Role, and similar objects in a consistent way (obviating the need for a custom HTML template)
2. define the column as being based on `installed_device__status`, which django-tables2 knows to mean that by default it uses that field lookup (i.e. `record.installed_device.status`) to provide the `value` for rendering this column.
    - This is necessary in combination with item 1, because by default a column called `status` would try to derive its value from `record.status`, which isn't a field on a DeviceBay record.
    - This fixes the original reported issue, in that now that this column natively understands its own `value`, django-tables2 can now intelligently sort the table based on this column's `value`.

There are absolutely use cases for TemplateColumn to handle various other advanced scenarios, so a blanket rule of "don't use TemplateColumn" wouldn't be useful. It would have to be something able to detect that:
1. You're using a TemplateColumn whose template is doing a nested lookup to get the actual value of interest
2. Using an appropriate column name (or specifying an `accessor` on the column definition) would let you get the actual value more effectively than doing the lookup in the template code itself.